fs/romfs/fs_romfs.c

```diff
@@ -617,6 +608,35 @@ static int romfs_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
   return -ENOTTY;
 }
 
+static int     romfs_mmap(FAR struct file *filep, FAR void **addr,
+                          off_t start, size_t length)
+{
+  FAR struct romfs_mountpt_s *rm;
+  FAR struct romfs_file_s    *rf;
+  int ret = -EINVAL;
+
+  /* Sanity checks */
+
+  DEBUGASSERT(filep->f_priv != NULL && filep->f_inode != NULL);
+
+  /* Recover our private data from the struct file instance */
+
+  rf = filep->f_priv;
+  rm = filep->f_inode->i_private;
+
+  /* Return the address on the media corresponding to the start of
```

Comment:
let's check offset/length inside file

Reply:
It is not immediately obvious to me how the thing works; I'll leave this improvement for later.